code/src/beetle.cpp

```diff
@@ -100,8 +100,91 @@ namespace r2d2::moving_platform {
                 counter_m1 > (encode_1_full_turn * turn / 360 * degrees)) {
                 break;
             }
-            // wait so the while loop aint blocking
-            hwlib::wait_ms(0.1);
+        }
+    }
+    void beetle_c::move(int16_t distance) {
+        // wheel has a circumference of 39 cm
+        // Encode frequency for 1 turn of the wheel. the encoder has 64 point
+        // per over 2 pins we count when de adc goes from low to high of 1 pin.
+        // The gear ratio from the motor is 50:1 64/4*50 = 800
```

Comment:
maybe add the 39cm of the wheel and the total rotations(800) and the 3cm stopping distance to the private variables asswell this way the values can be shared between this function and the turn function since the total rotations is used here asswell. also this makes the code more future proof since if an other wheel is mounted or gear ratio etc. is changed only one variable needs adjustment same goes voor stopping distance this will result in more future proofing and more readble equations on line 112 for example